client/client-lib/src/api.rs

```diff
@@ -345,6 +365,28 @@ impl<C: JsonRpcClient + Debug + Send + Sync> IFederationApi for WsFederationApi<
             Err(e) => Err(e),
         }
     }
+
+    async fn upload_ecash_backup(&self, request: &SignedBackupRequest) -> Result<()> {
+        self.request(
+            "/mint/backup",
+            request,
+            CurrentConsensus::new(self.peers().threshold()),
+        )
+        .await
+    }
+
+    async fn download_ecash_backup(&self, id: &secp256k1::XOnlyPublicKey) -> Result<Vec<u8>> {
+        let hex_str: String = self
+            .request(
+                "/mint/recover",
+                id,
+                // TODO: do we need a different strategy for this?
+                TrustAllPeers,
```

Comment:
Good point! It's not a DoS vector since you can technically even recover without the backup. But it will take a long time. So using union and then taking the most up-to-date one that's signed correctly is probably the better strategy.